Modix.Data/Repositories/MessageRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Models.Core;
+
+namespace Modix.Data.Repositories
+{
+    public interface IMessageRepository
+    {
+        Task<IReadOnlyCollection<MessageEntity>> GetRecentUserMessagesAsync(ulong guildId, ulong userId, TimeSpan timespan);
+
+        Task AddAsync(MessageEntity message);
```

Comment:
Should be `CreateAsync` for consistency.

Reply:
Also, should be `MessageCreationData`

Reply:
I agree with the naming here, though I don't think we necessarily need a CreationData model if it won't actually be any different

Reply:
Agreed, fixed in 31374be.